staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go

```diff
@@ -240,18 +241,23 @@ func newJSONPathParser(jsonPathExpression string) (*jsonpath.JSONPath, error) {
 	return j, nil
 }
 
-// processJSONPathInput will parses the user's JSONPath input and process the string
-func processJSONPathInput(jsonPathExpression, jsonPathCond string) (string, string, error) {
-	relaxedJSONPathExp, err := cmdget.RelaxedJSONPathExpression(jsonPathExpression)
+// processJSONPathInput will parses the user's JSONPath input containing JSON expression and, optionally, JSON value for matching condition and process it
+func processJSONPathInput(jsonPathInput []string) (string, string, error) {
+	if numOfArgs := len(jsonPathInput); numOfArgs < 1 || numOfArgs > 2 {
+		return "", "", fmt.Errorf("jsonpath wait format must be --for=jsonpath='{.status.readyReplicas}'=3 or --for=jsonpath='{.status.readyReplicas}'")
+	}
+	relaxedJSONPathExp, err := cmdget.RelaxedJSONPathExpression(jsonPathInput[0])
 	if err != nil {
 		return "", "", err
 	}
-	if jsonPathCond == "" {
-		return "", "", errors.New("jsonpath wait condition cannot be empty")
+	if len(jsonPathInput) > 1 {
+		jsonPathValue := jsonPathInput[1]
+		if jsonPathValue == "" {
+			return "", "", errors.New("jsonpath wait value cannot be empty")
```

Comment:
Could you please give an example of how `jsonPathInput[1]` is empty?. Because we checked that length of jsonPathInput is 1 or 2 already.

Reply:
I've rewrote the error message to explain how the value can be empty